test/extended/util/framework.go

```diff
@@ -236,127 +236,152 @@ func processScanError(log string) error {
-							}
-							isi := &imagev1.ImageStreamImport{}
-							isi.Name = lang
-							isi.Namespace = "openshift"
-							isi.ResourceVersion = stream.ResourceVersion
-							isi.Spec = imagev1.ImageStreamImportSpec{
-								Import: true,
-								Images: []imagev1.ImageImportSpec{},
-							}
-							for _, tag := range stream.Spec.Tags {
-								if tag.From != nil && tag.From.Kind == "DockerImage" {
-									iis := imagev1.ImageImportSpec{}
-									iis.From = *tag.From
-									iis.To = &corev1.LocalObjectReference{Name: tag.Name}
-									isi.Spec.Images = append(isi.Spec.Images, iis)
-								}
-							}
-							_, err = oc.AsAdmin().ImageClient().ImageV1().ImageStreamImports("openshift").Create(ctx, isi, metav1.CreateOptions{})
-							if err != nil {
-								return processScanError(fmt.Sprintf("after seeing FailedImageImports for %s the manual image import failed with %s", lang, err.Error()))
-							}
-							return processScanError(fmt.Sprintf("after seeing FailedImageImports for %s a manual image-import was submitted", lang))
+			return false, err
+		}
+		if retried {
+			return false, nil
+		}
+		return checkOpenShiftNamespaceImageStreamImported(ctx, oc, imagestream, registryHostname)
+	})
+	// pollErr will be not nil if there was an immediate error, or we timed out.
+	if pollErr == nil {
+		return nil
+	}
+	DumpImageStream(oc, "openshift", imagestream)
+	// If retried=true at this point, it means that we have repeatedly tried to reimport the imagestream and failed to do so.
+	// This could be an indicator that the Red Hat Container Registry (registry.redhat.io) is experiencing an outage, since most samples operator imagestream images are hosted there.
+	if retried {
+		strbuf := bytes.Buffer{}
+		strbuf.WriteString("Failed immagestream imports may indicate an issue with the Red Hat Container Registry (registry.redhat.io).\n")
+		strbuf.WriteString(" - check status at https://status.redhat.com (catalog.redhat.com) for reported outages\n")
+		strbuf.WriteString(" - if no outages are reported there, email Terms-Based-Registry-Team@redhat.com with a report of the error\n")
+		strbuf.WriteString("   and prepare to work with the test platform team to get the current set of tokens for CI\n")
+		e2e.Logf(strbuf.String())
+	}
+	return pollErr
+}
+
+// retrySamplesOperatorImagestreamImport immediately retries an import for the provided imagestream if an imagestream managed by the samples operator fails to import.
+// This allows imagestreams to be reimported at a faster cadence than what the samples operator currently provides.
+// Imagestreams which use images in the OCP payload are not managed by the samples operator and therefore will not be retried.
+// Returns true if the imagestream import was retried.
+func retrySamplesOperatorImagestreamImport(ctx context.Context, oc *CLI, imagestream string) (bool, error) {
```

Comment:
I understand the name of the bool param could indicate that retries may not occur, but I think that is still somewhat subtle.

I also think we can lose `Operator` from the name.

I'd perfer `retryAsNeededSampleImagestreamImport`